components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/ProxyServiceResource.java

```diff
@@ -51,26 +61,42 @@ public ProxyServiceResource(String urlTemplate){
     public Set<String> getMethods() {
         Set<String> methods = new HashSet<>();
         methods.add(Constants.HTTP_GET);
+        methods.add(Constants.HTTP_POST);
         return methods;
     }
 
     @Override
     public boolean invoke(MessageContext messageContext) {
 
         buildMessage(messageContext);
-
         org.apache.axis2.context.MessageContext axis2MessageContext =
                 ((Axis2MessageContext) messageContext).getAxis2MessageContext();
 
-        String param = Utils.getQueryParameter(messageContext, "proxyServiceName");
-
-        if (Objects.nonNull(param)) {
-            populateProxyServiceData(messageContext, param);
+        if (messageContext.isDoingGET()) {
+            String param = Utils.getQueryParameter(messageContext, "proxyServiceName");
+            if (Objects.nonNull(param)) {
+                populateProxyServiceData(messageContext, param);
+            } else {
+                populateProxyServiceList(messageContext);
+            }
+            axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
         } else {
```

Comment:
I suppose the else condition is assuming a POST request? What if we get a DELETE/PUT request? Do we not need to respond with a valid message saying these operations are not valid.

Reply:
IIRC the not supported method calls will not propagate to here, it will be responded back in mediation ( carbon-mediation) level itself.

Reply:
As nirothipan has mentioned, this resource will not receive any requests other than GET and POST.